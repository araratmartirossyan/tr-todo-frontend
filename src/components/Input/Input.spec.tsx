import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { StyledInput } from './index'

interface StateOfTest {
  mockCallback?: jest.Mock
}

describe('Input test', () => {
  const returnState = ({ mockCallback = jest.fn() }: StateOfTest) =>
    shallow(<StyledInput name="test" onInput={mockCallback} />)

  it('should render component', () => {
    const component = returnState({})
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should switch view on click', () => {
    const mockCallback = jest.fn()
    const component = returnState({ mockCallback })
    component.simulate('input')
    expect(mockCallback.mock.calls.length).toBe(1)
  })
})
