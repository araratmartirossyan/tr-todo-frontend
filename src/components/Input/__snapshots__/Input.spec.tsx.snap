// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test should render component 1`] = `
<input
  className="sc-bdnxRM gXlJqt"
  name="test"
  onInput={[MockFunction]}
/>
`;
